lib/parts.lua

```diff
@@ -108,7 +111,11 @@ function parts:whole_notes(phase)
 
   if song.measure == 8 then self.drumPat = 2 
   elseif song.measure == 16 then self.drumPat = 3
-  elseif song.measure == 24 then self.drumPat = 4 end
+  elseif song.measure == 24 then self.drumPat = 4 
+  elseif song.measure >= parts.endingMeas then  
+    song.is_screen_dirty = true
+    song.is_playing = false 
```

Comment:
i'd recommend swapping these two lines for two reasons:
1. it will make it clear to future developers (probably you!) that setting `is_playing` to false then requires a redraw. as is, that is not apparent.
2. i'm not a lua expert, but i think there is a chance that the redraw will happen *between* these two lines... meaning that the redraw will happen, the the song will stop playing.

Reply:
did you mean to add `resetSong()` here?

Reply:
> i'd recommend swapping these two lines for two reasons:
> 
>     1. it will make it clear to future developers (probably you!) that setting `is_playing` to false then requires a redraw. as is, that is not apparent.
> 
>     2. i'm not a lua expert, but i think there is a chance that the redraw will happen _between_ these two lines... meaning that the redraw will happen, the the song will stop playing.

ooh, great catch. thank you.

> did you mean to add `resetSong()` here?

for the 'thank you' message to be readable without disappearing unexpectedly, i've only put `resetSong()` in `key()` and added instructions on how to start the song again in the 'thank you' message.